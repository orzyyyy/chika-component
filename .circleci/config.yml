version: 2

references:
  container_config: &container_config
    docker:
      - image: circleci/node:11.0.0
    working_directory: ~/chika-component

  attach_workspace: &attach_workspace
    attach_workspace:
      at: ~/chika-component

jobs:
  setup:
    <<: *container_config
    steps:
      - checkout
      - run: node -v
      - run: npm -v
      - run: npm install
      - run:
          command: |
            set +eo
            npm ls
            true
      - persist_to_workspace:
          root: ~/chika-component
          paths:
            - node_modules
  lint:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run lint
  test:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run codecov
      - run: bash <(curl -s https://codecov.io/bash) -t 23446e71-0065-4582-b354-266b384f3625
  compile:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run compile
      - run: ls lib
      - run: ls lib/component
      - run: node ./scripts/dekko-lib.js
      - persist_to_workspace:
          root: ~/chika-component
          paths:
            - lib
  test_lib:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run:
          command: npm test -- --clearCache -u
          environment:
            LIB_DIR: lib

workflows:
  version: 2
  build:
    jobs:
      - setup
      - test:
          requires:
            - setup
      - lint:
          requires:
            - setup
      - compile:
          requires:
            - setup
      - test_lib:
          requires:
            - compile
