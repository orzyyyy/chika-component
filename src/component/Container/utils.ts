export const formatConfig = (config: Array<any>) => {
  let result: any = [];
  for (let item of config) {
    const {
      fname,
      controltype,
      fvalue,
      isvisiable,
      isadd,
      defaultvalue,
      editpageorderid,
      isnull,
      issearchfield,
      maxlen,
      minlen,
      foreigndata,
    } = item;
    result.push({
      type: (controlTypeEnums as any)[controltype],
      key: fname,
      name: fvalue,
      showInEdit: isadd,
      showInDetail: isvisiable,
      defaultValue: defaultvalue,
      orderInEdit: editpageorderid,
      orderInDetail: editpageorderid,
      isNull: isnull,
      isSearchItem: issearchfield,
      maxLength: maxlen,
      minLength: minlen,
      foreignData: foreigndata,
    });
  }
  return result;
};

const controlTypeEnums = {
  1: 'input',
  2: 'datePicker',
  3: 'select',
  5: 'checkbox',
  9: 'calendar',
  12: 'upload',
  14: 'mapPicker',
  99: 'label',
};

export const formatControls = (
  dataItem: any,
  config: Array<any>,
  primaryKey: string,
) => {
  let result = [];
  const keys = Object.keys(dataItem);
  for (let item of keys) {
    const targetItem = config.filter(target => target.key === item);
    if (targetItem.length) {
      const { key, showInDetail, type } = targetItem[0];
      if (!showInDetail) continue;
      const value = dataItem[key];
      const item = {
        ...targetItem[0],
        value: dataItem[key],
        templateOrder: dataItem.templateOrder,
        // when mapPicker change, dataSource will change target item by this.
        primaryValue: dataItem[primaryKey],
      };

      // handle with mapPicker
      if (type === 'mapPicker') {
        const latng = value.split('|');
        result.push({
          ...item,
          lng: latng[0],
          lat: latng[1],
          address: latng[2],
        });
      } else {
        result.push(item);
      }
    }
  }
  return result;
};
