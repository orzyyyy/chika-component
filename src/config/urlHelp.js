define([], function () {
  var urlHelp = {
    isReal: false,
    real: {
      search: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getdata',
      getConfig: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getconfig',
      getUrlData: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getinterfacedata', // 第三方接口地址，为了放置跨域问题，后端先处理下
      generalbackstage: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/operatedata', // 增删改
    },
    simulation: {
      tableconfig: './data/tableconfig.json',
      tableDatas: './data/tableDatas.json',
      newTabDatas: './data/newTabDatas.json',
      builder_search: '../../CommonReport/AjaxHandler/BuilderHandler.ashx?OPType=Search',
      getConfig: './data/getConfig.json',
      search: './data/search.json',
      calendar_demo: './data/calendar_demo.json',
      secRecordPath: './data/s_problem_record_hy.json',
      getUrlData: './data/tableconfig.json', // 第三方接口地址，为了防止跨域问题，后端先处理下
      generalbackstage: './data/tableconfig.json', // 增删改
    }
  }
  return urlHelp
});