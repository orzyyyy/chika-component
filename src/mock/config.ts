export const originConfig = [
  {
    id: 113442,
    fname: 'pjnm',
    fvalue: 'pjnm',
    physicalcolname: '',
    iskey: false,
    isnull: false,
    issearchfield: true,
    searchtype: -1,
    searchorder: null,
    minlen: 0,
    maxlen: 5,
    decimalcount: null,
    regular: '',
    orderid: 26,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 99,
    editpageorderid: 21,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
    fieldpar: '',
  },
  {
    id: 113438,
    fname: 'dam_cd',
    fvalue: 'dam_cd',
    physicalcolname: 'dam_cd',
    iskey: true,
    isnull: false,
    issearchfield: true,
    searchtype: 8,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 1,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 1,
    editpageorderid: 1,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
    fieldpar: '',
    isSort: true,
  },
  {
    id: 113442,
    fname: 'datePicker',
    fvalue: 'datePicker',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 50,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 2,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'select',
    fvalue: 'select',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 3,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [
      { label: 'label1', value: 'value1' },
      { label: 'label2', value: 'value2' },
    ],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'checkbox',
    fvalue: 'checkbox',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 5,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [
      { label: 'checkboxlable1', value: 'checkbox1' },
      { label: 'label2', value: 'checkbox2' },
      { label: 'label3', value: 'checkbox3' },
    ],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'calendar',
    fvalue: 'calendar',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 9,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'upload',
    fvalue: 'upload',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: 2,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 12,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'mapPicker',
    fvalue: 'mapPicker',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: 2,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 14,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
];
