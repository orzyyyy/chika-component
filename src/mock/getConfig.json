{
  "data": {
    "power": "Select,Add,Del,Update",
    "tablefieldconfig": [
      {
        "id": 113442,
        "fname": "pjnm",
        "fvalue": "工程名称",
        "physicalcolname": "",
        "iskey": false,
        "isnull": false,
        "issearchfield": true,
        "searchtype": -1,
        "searchorder": null,
        "minlen": null,
        "maxlen": null,
        "decimalcount": null,
        "regular": "",
        "orderid": 26,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "",
        "controltype": 99,
        "editpageorderid": 21,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": ""
      },
      {
        "id": 113438,
        "fname": "dam_cd",
        "fvalue": "堰坝名称代码",
        "physicalcolname": "dam_cd",
        "iskey": true,
        "isnull": false,
        "issearchfield": true,
        "searchtype": 8,
        "searchorder": null,
        "minlen": null,
        "maxlen": 12,
        "decimalcount": null,
        "regular": "",
        "orderid": 1,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "",
        "controltype": 1,
        "editpageorderid": 1,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": ""
      },
      {
        "id": 113439,
        "fname": "retain_dam_type",
        "fvalue": "挡水主坝类型",
        "physicalcolname": "retain_dam_type",
        "iskey": false,
        "isnull": false,
        "issearchfield": true,
        "searchtype": 8,
        "searchorder": null,
        "minlen": null,
        "maxlen": 10,
        "decimalcount": null,
        "regular": "",
        "orderid": 6,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "",
        "controltype": 1,
        "editpageorderid": 6,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": ""
      },
      {
        "id": 113440,
        "fname": "dam_width",
        "fvalue": "坝高",
        "physicalcolname": "dam_width",
        "iskey": false,
        "isnull": false,
        "issearchfield": true,
        "searchtype": 5,
        "searchorder": null,
        "minlen": null,
        "maxlen": 9,
        "decimalcount": 3,
        "regular": "",
        "orderid": 11,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "m",
        "controltype": 1,
        "editpageorderid": 11,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": ""
      },
      {
        "id": 113441,
        "fname": "crest_length",
        "fvalue": "坝长",
        "physicalcolname": "crest_length",
        "iskey": false,
        "isnull": false,
        "issearchfield": true,
        "searchtype": 5,
        "searchorder": null,
        "minlen": null,
        "maxlen": 8,
        "decimalcount": 2,
        "regular": "",
        "orderid": 21,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "m",
        "controltype": 1,
        "editpageorderid": 16,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": "{\r\n    IsUploadMultFile:\"y\",/*是否为多附件上传，y为是，其他或不填单附件*/\r\n    FileListHeight:80,/*上传后附件展示的列表的高度，这边填数字*/\r\n    FileListWidth:\"\",/*上传后附件展示的列表的宽度，这边填数字*/\r\n    UploadFileLimit:\"20971520\",/*文件上传最大长度默认为20MB：20*1024*1024=20971520*/\r\n    IsHideDelBtn:\"false\",/*是否因此删除按钮，默认是隐藏的，为false的时候才显示出来*/\r\n    UploadType:\"*.doc;*.docx;*.xls;*.xlsx;*.ppt;*.pptx;*.mpp;*.vsd\",/*文件类型*/\r\n}"
      },
      {
        "id": 113442,
        "fname": "latng",
        "fvalue": "地址",
        "physicalcolname": "latng",
        "iskey": false,
        "isnull": false,
        "issearchfield": false,
        "searchtype": 5,
        "searchorder": null,
        "minlen": null,
        "maxlen": 8,
        "decimalcount": 2,
        "regular": "",
        "orderid": 21,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "",
        "controltype": 14,
        "editpageorderid": 16,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": "",
        "fieldpar": "{\"lng\":\"lng1\",\"lat\":\"lat1\",\"address\": \"address1\"}"
      },
      {
        "id": 113442,
        "fname": "attachment",
        "fvalue": "日期",
        "physicalcolname": "attachment",
        "iskey": false,
        "isnull": false,
        "issearchfield": false,
        "searchtype": 5,
        "searchorder": null,
        "minlen": null,
        "maxlen": 8,
        "decimalcount": 2,
        "regular": "",
        "orderid": 21,
        "isvisiable": true,
        "defaultvalue": "",
        "unit": "",
        "controltype": 2,
        "editpageorderid": 16,
        "controlheight": "",
        "controlwidth": "",
        "isadd": true,
        "fieldtag": null,
        "foreigndata": [],
        "dateformat": ""
      }
    ]
  }
}
