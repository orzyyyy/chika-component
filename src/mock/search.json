{
  "data": {
    "list": [
      {
        "rownum": 1,
        "dam_cd": "Z000001",
        "dam_width": 16,
        "crest_length": 63,
        "retain_dam_type": "堆石坝",
        "pjnm": "德江坞",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州1",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 2,
        "dam_cd": "Z000002",
        "dam_width": 0.7,
        "crest_length": 13.8,
        "retain_dam_type": "堆石坝",
        "pjnm": "六谷地村混凝土堰坝1",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox3",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 3,
        "dam_cd": "Z000003",
        "dam_width": 2.4,
        "crest_length": 42.4,
        "retain_dam_type": "混凝土坝",
        "pjnm": "赤龙坝",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox3",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 4,
        "dam_cd": "Z000004",
        "dam_width": 3.5,
        "crest_length": 45,
        "retain_dam_type": "混凝土坝",
        "pjnm": "湖家埭坝2",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox2",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 5,
        "dam_cd": "Z000005",
        "dam_width": 6.2,
        "crest_length": 34,
        "retain_dam_type": "混凝土坝",
        "pjnm": "家猫坞山塘",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox2",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 6,
        "dam_cd": "Z000006",
        "dam_width": 16,
        "crest_length": 66,
        "retain_dam_type": "混凝土坝",
        "pjnm": "南塘坞山塘",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 7,
        "dam_cd": "Z000007",
        "dam_width": 12.1,
        "crest_length": 38,
        "retain_dam_type": "混凝土坝",
        "pjnm": "西步坞山塘",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 8,
        "dam_cd": "Z000008",
        "dam_width": 0.9,
        "crest_length": 22,
        "retain_dam_type": "混凝土坝",
        "pjnm": "东沈村混凝土堰坝1",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 9,
        "dam_cd": "Z000009",
        "dam_width": 1.6,
        "crest_length": 17.3,
        "retain_dam_type": "混凝土坝",
        "pjnm": "东沈村混凝土堰坝2",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      },
      {
        "rownum": 10,
        "dam_cd": "Z000010",
        "dam_width": 1.2,
        "crest_length": 15,
        "retain_dam_type": "混凝土坝",
        "pjnm": "东沈村混凝土堰坝3",
        "mapPicker": "116.30038976669313|39.89520118467971|杭州",
        "datePicker": "2019-03-11",
        "select": "value1",
        "checkbox": "checkbox1",
        "calendar": "2018-04-03 00:00:00,2018-04-04 00:00:00",
        "upload": "upload"
      }
    ],
    "recordcount": 21
  }
}
