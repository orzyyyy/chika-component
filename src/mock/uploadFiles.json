{
  "data": {
    "list": [
      {
        "id": 1,
        "annex": "[{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name0\",\"filename\":\"xls\"},{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name1\",\"filename\":\"doc\"}]"
      },
      {
        "id": 2,
        "annex": "[{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name2\",\"filename\":\"jpg\"}]"
      }
    ],
    "recordcount": 3
  }
}
