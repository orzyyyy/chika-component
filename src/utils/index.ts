import { format as fnsFormat } from 'date-fns/esm';
import { zhCN } from 'date-fns/locale';

export const formatDate = (
  date: Date | string = new Date(),
  format: string = 'yyyy-MM-dd HH:mm:ss',
) => {
  if (typeof date === 'string') {
    date = new Date(date);
  }
  return fnsFormat(date, format, {
    locale: zhCN,
  });
};

export const getParamsFromUrl = (name: string, url?: string) => {
  const targetUrl = url || window.location.hash;
  const reg = new RegExp('(^|&)' + name + '=([^&]*)(&|$)');
  let result: any =
    targetUrl.split('?').length > 1 ? targetUrl.split('?')[1].match(reg) : 0;
  if (result) {
    return result[2];
  }
  return null;
};

export const compare = (property: any, isDesc?: boolean) =>
  isDesc
    ? (b: any, a: any) => a[property] - b[property]
    : (a: any, b: any) => a[property] - b[property];

export const getDirection = (
  startX: number,
  startY: number,
  endX: number,
  endY: number,
) => {
  let angx = endX - startX;
  let angy = endY - startY;

  if (Math.abs(angx) < 25 && Math.abs(angy) < 25) {
    return '';
  }

  let angle = (Math.atan2(angy, angx) * 180) / Math.PI;

  if (angle >= -135 && angle <= -45) {
    return 'toTop';
  } else if (angle > 45 && angle < 135) {
    return 'toDown';
  } else if (
    (angle >= 135 && angle <= 180) ||
    (angle >= -180 && angle < -135)
  ) {
    return 'toLeft';
  } else if (angle >= -45 && angle <= 45) {
    return 'toRight';
  }
  return '';
};

export const bindTouchDirection = (
  ref: any,
  callback: (direction: string) => void,
) => {
  if (!ref) {
    return callback('error');
  }
  let startX: number;
  let startY: number;
  let endX: number;
  let endY: number;

  ref.addEventListener('touchstart', (e: any) => {
    startX = e.touches[0].pageX;
    startY = e.touches[0].pageY;
  });

  ref.addEventListener('touchend', (e: any) => {
    endX = e.changedTouches[0].pageX;
    endY = e.changedTouches[0].pageY;

    callback(getDirection(startX, startY, endX, endY));
  });
};
