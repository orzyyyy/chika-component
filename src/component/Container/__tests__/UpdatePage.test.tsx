import React from 'react';
import { mount, shallow } from 'enzyme';
import 'nino-cli/scripts/setup';
import { originConfig } from '../../../mock/config';
// import { originDataSource } from '../../../mock/dataSource';
import { formatControls, formatConfig } from '../utils';
const Config = formatConfig(originConfig);
let UpdatePage;
switch (process.env.LIB_DIR) {
  case 'lib':
    UpdatePage = require('../../../../lib/component/Container/UpdatePage')
      .default;
    break;
  default:
    UpdatePage = require('../UpdatePage').default;
    break;
}

describe('UpdatePage', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <UpdatePage
        dataSource={formatControls(null, Config, 'dam_cd')}
        status="add"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is add', () => {
    const wrapper = mount(
      <UpdatePage
        dataSource={formatControls(null, Config, 'dam_cd')}
        status="add"
      />,
    );
    const listContent = wrapper.find('.am-list-content');
    expect(listContent.length).toBe(9);
    expect(listContent.at(0).text()).toBe('pjnm');
    expect(listContent.at(1).text()).toBe('dam_cd');
    expect(listContent.at(2).text()).toBe('datePicker');
    expect(listContent.at(3).text()).toBe('select');
    expect(listContent.at(4).text()).toBe('calendar');
    expect(listContent.at(5).text()).toBe('起始时间');
    expect(listContent.at(6).text()).toBe('结束时间');
    expect(listContent.at(7).text()).toBe('mapPicker');
  });

  it('should update state correctly when status is add', () => {
    const dataSource = formatControls(null, Config, 'dam_cd');
    const wrapper = mount(<UpdatePage dataSource={dataSource} status="add" />);
    let transformedDataSource = [];
    for (let item of dataSource) {
      transformedDataSource.push({
        ...item,
        error: false,
        message: '',
      });
    }
    expect(wrapper.state().form).toEqual(transformedDataSource);
    // input
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { target: { value: 'test' } });
    expect(wrapper.state().form[1].value).toBe('test');
    // datePicker
    wrapper
      .find('ListItem')
      .at(2)
      .simulate('click');
    wrapper.find('.am-picker-popup-header-right').simulate('click');
    expect(wrapper.state().form[2].value.getDate()).toBe(new Date().getDate());
    // select
    wrapper
      .find('ListItem')
      .at(3)
      .simulate('click');
    wrapper.find('.am-picker-popup-header-right').simulate('click');
    expect(wrapper.state().form[3].value).toBe('value1');
    // checkbox
    wrapper
      .find('.am-accordion-header')
      .first()
      .simulate('click');
    wrapper
      .find('.am-checkbox-input')
      .at(2)
      .simulate('change');
    expect(wrapper.state().form[4].value).toBe('checkbox3');
    // calendar
    // upload
    // mapPicker
    // they are hard to test
  });
});
