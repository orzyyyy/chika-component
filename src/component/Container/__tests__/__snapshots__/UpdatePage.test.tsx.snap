// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePage render correctly 1`] = `
<Fragment>
  <List
    prefixCls="am-list"
  >
    <ListItem
      align="middle"
      error={false}
      extra={
        <InputItem
          clear={true}
          disabled={false}
          editable={true}
          error={false}
          extra=""
          labelNumber={5}
          moneyKeyboardAlign="right"
          moneyKeyboardHeader={null}
          moneyKeyboardWrapProps={Object {}}
          onBlur={[Function]}
          onChange={[Function]}
          onErrorClick={[Function]}
          onExtraClick={[Function]}
          onFocus={[Function]}
          onVirtualKeyboardConfirm={[Function]}
          placeholder="请输入"
          prefixCls="am-input"
          prefixListCls="am-list"
          style={
            Object {
              "textAlign": "right",
            }
          }
          type="text"
          updatePlaceholder={false}
          value=""
        />
      }
      key="update-page-add-input-item-113442"
      multipleLine={false}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      pjnm
    </ListItem>
    <ListItem
      align="middle"
      error={false}
      extra={
        <InputItem
          clear={true}
          disabled={false}
          editable={true}
          error={false}
          extra=""
          labelNumber={5}
          moneyKeyboardAlign="right"
          moneyKeyboardHeader={null}
          moneyKeyboardWrapProps={Object {}}
          onBlur={[Function]}
          onChange={[Function]}
          onErrorClick={[Function]}
          onExtraClick={[Function]}
          onFocus={[Function]}
          onVirtualKeyboardConfirm={[Function]}
          placeholder="请输入"
          prefixCls="am-input"
          prefixListCls="am-list"
          style={
            Object {
              "textAlign": "right",
            }
          }
          type="text"
          updatePlaceholder={false}
          value=""
        />
      }
      key="update-page-add-input-item-113438"
      multipleLine={false}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      dam_cd
    </ListItem>
    <DatePicker
      key="update-page-add-data-picker-113442"
      minuteStep={1}
      mode="datetime"
      onChange={[Function]}
      pickerPrefixCls="am-picker-col"
      popupPrefixCls="am-picker-popup"
      prefixCls="am-picker"
      use12Hours={false}
      value=""
    >
      <ListItem
        align="middle"
        arrow="horizontal"
        error={false}
        multipleLine={false}
        platform="ios"
        prefixCls="am-list"
        wrap={false}
      >
        datePicker
      </ListItem>
    </DatePicker>
    <Picker
      cascade={true}
      cols={1}
      data={
        Array [
          Object {
            "label": "label1",
            "value": "value1",
          },
          Object {
            "label": "label2",
            "value": "value2",
          },
        ]
      }
      format={[Function]}
      key="update-page-add-select-113442"
      onChange={[Function]}
      pickerPrefixCls="am-picker-col"
      popupPrefixCls="am-picker-popup"
      prefixCls="am-picker"
      title=""
      triggerType="onClick"
      value={Array []}
    >
      <ListItem
        align="middle"
        arrow="horizontal"
        error={false}
        multipleLine={false}
        platform="ios"
        prefixCls="am-list"
        wrap={false}
      >
        select
      </ListItem>
    </Picker>
    <Accordion
      key="update-page-add-checkbox-113442"
      prefixCls="am-accordion"
    >
      <CollapsePanel
        destroyInactivePanel={false}
        forceRender={false}
        header="checkbox"
        headerClass=""
        isActive={false}
        onItemClick={[Function]}
        showArrow={true}
      >
        <List
          prefixCls="am-list"
        >
          <CheckboxItem
            checkboxProps={Object {}}
            checked={false}
            key="update-page-add-checkbox-item-0"
            listPrefixCls="am-list"
            onChange={[Function]}
            prefixCls="am-checkbox"
          >
            checkboxlable1
          </CheckboxItem>
          <CheckboxItem
            checkboxProps={Object {}}
            checked={false}
            key="update-page-add-checkbox-item-1"
            listPrefixCls="am-list"
            onChange={[Function]}
            prefixCls="am-checkbox"
          >
            label2
          </CheckboxItem>
          <CheckboxItem
            checkboxProps={Object {}}
            checked={false}
            key="update-page-add-checkbox-item-2"
            listPrefixCls="am-list"
            onChange={[Function]}
            prefixCls="am-checkbox"
          >
            label3
          </CheckboxItem>
        </List>
      </CollapsePanel>
    </Accordion>
    <ListItem
      align="middle"
      arrow="horizontal"
      error={false}
      extra="请选择"
      multipleLine={false}
      onClick={[Function]}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      calendar
    </ListItem>
    <ListItem
      align="middle"
      error={false}
      multipleLine={false}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      起始时间
    </ListItem>
    <ListItem
      align="middle"
      error={false}
      multipleLine={false}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      结束时间
    </ListItem>
    <Accordion
      key="update-page-add-upload-113442"
      prefixCls="am-accordion"
    >
      <CollapsePanel
        destroyInactivePanel={false}
        forceRender={false}
        header="upload"
        headerClass=""
        isActive={false}
        onItemClick={[Function]}
        showArrow={true}
      >
        <Upload
          fileList=""
          onChange={[Function]}
        />
      </CollapsePanel>
    </Accordion>
    <ListItem
      align="middle"
      arrow="horizontal"
      error={false}
      extra="请选择"
      key="update-page-add-map-picker-add-113442"
      multipleLine={false}
      onClick={[Function]}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      mapPicker
    </ListItem>
    <ListItem
      align="middle"
      error={false}
      multipleLine={false}
      platform="ios"
      prefixCls="am-list"
      wrap={false}
    >
      <Button
        activeStyle={Object {}}
        disabled={false}
        inline={true}
        loading={false}
        prefixCls="am-button"
        size="large"
        style={
          Object {
            "marginRight": 4,
            "width": "calc(50% - 4px)",
          }
        }
        type="primary"
      >
        保存
      </Button>
      <Button
        activeStyle={Object {}}
        disabled={false}
        inline={true}
        loading={false}
        prefixCls="am-button"
        size="large"
        style={
          Object {
            "width": "50%",
          }
        }
      >
        返回
      </Button>
    </ListItem>
  </List>
  <Calendar
    onCancel={[Function]}
    onConfirm={[Function]}
    pickTime={true}
    prefixCls="am-calendar"
    timePickerPickerPrefixCls="am-picker-col"
    timePickerPrefixCls="am-picker"
    visible={false}
  />
</Fragment>
`;
